app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt

```diff
@@ -25,23 +28,21 @@ import org.mozilla.fenix.ext.showToolbar
 
 @SuppressWarnings("TooManyFunctions")
 class DeleteBrowsingDataFragment : Fragment(R.layout.fragment_delete_browsing_data) {
-    private lateinit var sessionObserver: SessionManager.Observer
+
     private lateinit var controller: DeleteBrowsingDataController
+    private var scope: CoroutineScope? = null
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         controller = DefaultDeleteBrowsingDataController(requireContext())
 
-        sessionObserver = object : SessionManager.Observer {
-            override fun onSessionAdded(session: Session) = updateTabCount()
-            override fun onSessionRemoved(session: Session) = updateTabCount()
-            override fun onSessionSelected(session: Session) = updateTabCount()
-            override fun onSessionsRestored() = updateTabCount()
-            override fun onAllSessionsRemoved() = updateTabCount()
+        scope = requireComponents.core.store.flowScoped(viewLifecycleOwner) { flow ->
```

Comment:
We usually create the scope in `onStart` and cancel it `onStop`. Mostly curious if you did this differently for a reason? We never cancel the returned scope here though. Should we `scope?.cancel` in `onDestroyView` or switch to `onStart/onStop`?

Reply:
I'll switch to `onStart/onStop`